include/interfaces/ALE_navier_stokes.h

```diff
@@ -298,29 +310,36 @@ energies_and_residuals(const typename DoFHandler<dim,spacedim>::active_cell_iter
 
 // pressure:
           auto p_test = fev[pressure].value(i,quad);
-//          auto q = fev[pressure].value(i,quad);
-
+          auto grad_p_test = fev[pressure].gradient(i,quad);
+          
+          // Preconditioner:
           residual[1][i] +=
             (
-              (1./nu)*p*p_test
+              p * p_test * J_ale
             )*JxW[quad];
 
-          residual[0][i] +=
+          residual[2][i] +=
             (
-              // time derivative term
-              rho*scalar_product( u_dot * J_ale , u_test )
-//
-              + scalar_product( grad_u * ( F_inv * ( u_old - d_dot ) ) * J_ale , u_test )
+              grad_p * grad_p_test * J_ale
+            )*JxW[quad];
+
+          // Navier Stokes:
+          res_u = J_ale * (
+// time derivative term
+            + rho * scalar_product( 
+              u_dot + grad_u * ( F_inv * ( u_old - d_dot ) ), u_test )
 //
-              + scalar_product( J_ale * sigma * Ft_inv, grad_u_test )
+            + scalar_product(sigma * Ft_inv, grad_u_test ))
 // divergence free constriant
-              - div_u * p_test
-// pressure term
-              - p * div_u_test
-// Impose armonicity of d and v=d_dot
-              + scalar_product( grad_d , grad_d_test )
-            )*JxW[quad];
+            - div_u * p_test;
+            // - scalar_product(J_ale * F_inv * u, grad_p_test); // TODO: muovere la divergenza a sinistra! se no c'é anche l'integrale sul bordo.
```

Comment:
@luca-heltai Is it possible to evaluate **scalar_product(div(J_ale * F_inv * u), p_test)**?